cmd/thanos/receive.go

```diff
@@ -760,6 +764,8 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 
 	cmd.Flag("tsdb.no-lockfile", "Do not create lockfile in TSDB data directory. In any case, the lockfiles will be deleted on next startup.").Default("false").BoolVar(&rc.noLockFile)
 
+	cmd.Flag("tsdb.max-exemplars", "Enables support for ingesting exemplars and set the maximum number that will be stored. 0 (or less) disables exemplars storage.").Default("0").IntVar(&rc.tsdbMaxExemplars)
```

Comment:
What does it mean with number?  Could you explain what the number actually is (i.e. the amount of examplars stored?) and what it does (will it drop things?) - IMO would be huge bonus to provide extra documentation to explain users how they should set this value and how they could calculate this.

Reply:
Yep, it is number of exemplars stored, and yes, it'll drop the oldest exemplars (AFAIR, I have to check on the Prometheus TSDB side).

> IMO would be huge bonus to provide extra documentation to explain users how they should set this value and how they could calculate this.

Yep, I agree. Maybe we can add a page around operating Receive (or whole Thanos stack in general) with exemplars.

Reply:
For now, I tried to make the flag description more descriptive.